sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -1049,35 +1109,14 @@ public void Set(object value)
                 case JsonDocument d:
                     Set(d.RootElement);
                     break;
+                case Stream:
+                    throw new NotSupportedException("Assigning a 'Stream' type is not supported.");
```

Comment:
Are there other types we want to prevent assigning because they wouldn't work well with JsonSerializer.Serialize()?

Reply:
I imagine lots e.g., atomic types like `Mutex`, but whether we need to guard against them only time will tell, I think. It'd be a huge list, but do we necessarily need to list them all? Not to mention it'd only be ones we know about. I think as far as "basic" types, `Stream` is a great starter to exclude.

Reply:
Per arch board discussion, we will create an allow-list instead and throw on any assignment of something not either on the allow-list or a POCO composed of types on the allow list.

Reply:
BTW, thank you for this, @heaths!  I will include Mutex on my list of test cases :)